packages/clerk-js/src/ui/utils/phoneUtils.ts

```diff
@@ -115,3 +116,34 @@ export function stringToFormattedPhoneString(str: string): string {
   const parsed = parsePhoneString(str);
   return `+${parsed.code} ${formatPhoneNumber(parsed.number, parsed.pattern, parsed.code)}`;
 }
+
+export const sortByPriority = (a: CountryEntry, b: CountryEntry) => {
+  if (a.priority > b.priority) {
+    return -1;
+  }
+  if (a.priority < b.priority) {
+    return 1;
+  }
+  return 0;
+};
+
+export function getLongestValidCountryCode(value: string) {
+  let result = '';
+  let countries = null;
+  const numberValue = value.replace('+', '');
+  const countryGroup = [];
+
+  for (let i = 4; i > 0; i--) {
+    const code = numberValue.slice(0, i);
+    countries = [...IsoToCountryMap.values()].filter(o => o.code === code);
+
+    if (countries.length) {
+      countryGroup.push(countries);
```

Comment:
🙃 Another, more idiomatic way to write this would be
```suggestion
      countryGroup.push(...countries);
```
that way you also avoid using `flat()` below